extension/demo/src/bg/background.js

```diff
@@ -1,66 +1,85 @@
 'use strict'
 
 import { sendMessage, addListener } from '../modules/messaging'
-import { API_BASE, STORAGE_KEY } from '../modules/constants'
+import { ACCESS_TOKEN, PAYLOAD, TTL, API_BASE_WEB } from '../modules/constants'
+import * as api from '../modules/api'
+import * as storage from '../modules/storage'
 
-function onPopupTriggerAuth () {
-  chrome.storage.local.get([STORAGE_KEY], results => {
-    if (!results[STORAGE_KEY]) {
-      chrome.tabs.create({
-        url: `${API_BASE}/oauth/ext`,
-        active: true
-      })
-    }
-  })
+async function onPopupTriggerAuth () {
+  const accessToken = await storage.get(ACCESS_TOKEN)
+  if (!accessToken) {
+    chrome.tabs.create({
+      url: `${API_BASE_WEB}/oauth/ext`,
+      active: true
+    })
+  }
 }
 
 function onPopupOpenApps () {
   chrome.tabs.create({
-    url: `${API_BASE}/apps`,
+    url: `${API_BASE_WEB}/apps`,
     active: true
   })
 }
 
 async function onToken (data) {
   const { accessToken, subscription } = data
-  let payload = {}
-  payload[STORAGE_KEY] = {
-    auth: {
-      accessToken: accessToken,
-      subscription: subscription
-    }
-  }
-  await chrome.storage.local.set(payload)
-  const isAuthenticated = !chrome.runtime.lastError
+  const isAuthenticated = await storage.set({ accessToken, subscription })
+  storage.get(ACCESS_TOKEN).then(console.log)
   sendMessage('popup-set-view', {
     isAuthenticated
   })
+  updatePayload(accessToken)
+
   return {
     isAuthenticated
   }
 }
 
+function timeToLive (expiresAt) {
+  return expiresAt - (Math.floor(Date.now() / 1000) + 3600)
+}
+
+async function updatePayload (accessToken) {
+  const payload = await storage.get(PAYLOAD)
+  if (!payload) {
+    const subscriptionStatus = await api.fetchSubscriptionStatus(accessToken)
+    console.log(subscriptionStatus)
+    if (subscriptionStatus.hasOwnProperty(PAYLOAD)) {
+      storage.set(subscriptionStatus)
+    }
+  } else {
+    const ttl = await storage.get(TTL)
+    if (ttl) {
+      let timeout = timeToLive(ttl)
+      console.log(timeout)
+      setTimeout(updatePayload.bind(null, accessToken), timeout)
```

Comment:
Another, slightly cleaner, option:
```suggestion
      const callback = updatePayload(accessToken)
      setTimeout(callback, timeout)
```

Reply:
Argument type Promise<void> is not assignable to parameter type TimerHandler   
Type Promise<void> is not assignable to type string | Function    
Type Promise<void> is not assignable to type Function 

🤔

Reply:
My bad, it should of course return a wrapper function. I've updated the suggestion.